services/github/src/handlers.js

```diff
@@ -617,7 +617,7 @@ async function jobHandler(message) {
       });
 
       if (!isCollaborator) {
-        if (message.payload.details['event.type'].startsWith('pull_request.opened')) {
+        if (message.payload.details['event.type'].startsWith('pull_request.opened') && (this.getRepoPolicy(defaultBranchYml) === 'collaborators')) {
```

Comment:
Looks good to me!  Let's say `!== 'collaborators_quiet'` so it's more clear what we're excluding here.  And @owlishDeveloper can do the review..